test/testunusedvar.cpp

```diff
@@ -715,6 +732,39 @@ class TestUnusedVar : public TestFixture {
             "}");
         ASSERT_EQUALS("", errout.str());
 
+        functionVariableUsage(
+            "int x[] = {0, 1, 3};\n"
+            "int func() {\n"
+            "   *x = 2;\n"
+            "   return 1;\n"
+            "}\n"
+            "class C {\n"
+            "public:\n"
+            "   C() : x(func()) {}\n"
+            "   int x;\n"
+            "};\n"
+            "void f() {\n"
+            "   C c;\n"
+            "}");
+        ASSERT_EQUALS("", errout.str());
+
+        functionVariableUsage(
+            "int x[] = {0, 1, 3};\n"
+            "int func() {\n"
+            "   *(x + 1) = 2;\n"
+            "   return 1;\n"
+            "}\n"
+            "class C {\n"
+            "public:\n"
+            "   C() : x(func()) {}\n"
+            "   int x;\n"
+            "};\n"
+            "void f() {\n"
+            "   C c;\n"
+            "}");
+        // TODO: update astutils.cpp:1544 isVariableChanged() - array change through pointer arithmetic is not handled
```

Comment:
as far as I see we do not write a warning here.. and also there should be no warning here?

Reply:
`#define TODO_ASSERT_EQUALS( WANTED , CURRENT , ACTUAL )`
We write a warning here, but should not.